packages/core/http/core-http-server-internal/src/lifecycle_handlers.ts

```diff
@@ -90,11 +92,16 @@ export const createCustomHeadersPreResponseHandler = (config: HttpConfig): OnPre
     csp: { header: cspHeader },
   } = config;
 
+  const reportingEndpointUrl = `${config.ssl ? 'https' : 'http'}://${config.host}:${config.port}${
+    config.basePath
+  }/internal/security/analytics/_record_violations`;
+
   return (request, response, toolkit) => {
     const additionalHeaders = {
       ...securityResponseHeaders,
       ...customResponseHeaders,
-      'Content-Security-Policy': cspHeader,
+      'Content-Security-Policy': `${cspHeader}; report-to ${REPORTING_ENDPOINT_NAME}`,
```

Comment:
For the purpose of this POC I am always adding report-to directive but we might want to make this configurable